ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchAlertsTool.java

```diff
@@ -0,0 +1,172 @@
+    @Setter
+    private Parser<?, ?> inputParser;
+    @Setter
+    private Parser<?, ?> outputParser;
+
+    public SearchAlertsTool(Client client) {
+        this.client = client;
+
+        // probably keep this overridden output parser. need to ensure the output matches what's expected
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        final String tableSortOrder = parameters.getOrDefault("sortOrder", "asc");
+        final String tableSortString = parameters.getOrDefault("sortString", "monitor_name.keyword");
+        final int tableSize = parameters.containsKey("size") ? Integer.parseInt(parameters.get("size")) : 20;
+        final int startIndex = parameters.containsKey("startIndex") ? Integer.parseInt(parameters.get("startIndex")) : 0;
+        final String searchString = parameters.getOrDefault("searchString", null);
+
+        // not exposing "missing" from the table, using default of null
+        final Table table = new Table(tableSortOrder, tableSortString, null, tableSize, startIndex, searchString);
+
+        final String severityLevel = parameters.getOrDefault("severityLevel", "ALL");
+        final String alertState = parameters.getOrDefault("alertState", "ALL");
+        final String monitorId = parameters.getOrDefault("monitorId", null);
+        final String alertIndex = parameters.getOrDefault("alertIndex", null);
+        @SuppressWarnings("unchecked")
+        final List<String> monitorIds = parameters.containsKey("monitorIds")
+            ? gson.fromJson(parameters.get("monitorIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> workflowIds = parameters.containsKey("workflowIds")
+            ? gson.fromJson(parameters.get("workflowIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> alertIds = parameters.containsKey("alertIds") ? gson.fromJson(parameters.get("alertIds"), List.class) : null;
+
+        GetAlertsRequest getAlertsRequest = new GetAlertsRequest(
+            table,
+            severityLevel,
+            alertState,
+            monitorId,
+            alertIndex,
+            monitorIds,
+            workflowIds,
+            alertIds
+        );
+
+        // create response listener
+        // stringify the response, may change to a standard format in the future
+        ActionListener<GetAlertsResponse> getAlertsListener = ActionListener.<GetAlertsResponse>wrap(response -> {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Alerts=[");
```

Comment:
Is there any change alert can be: `"Testing" Alert`

Reply:
See comment on 103. This may change later on when there is standard input/output

Reply:
I'm not sure if that's answering my question. My point is if there's any possibility to have a alert  like `"Testing" Alert`, then I think it will make the string: `'Alerts=[""Testing" Alert"]'`, which should break the code.

Reply:
The response will certainly change to include much more data and formatted in a different way than what it is currently doing. I hope that answers your question

Reply:
Additionally, alerting should not allow such a name to exist as that would break the standard followed by other plugins, such as AD